libraries/abstractions/transport/secure_sockets/transport_secure_sockets.h

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#ifndef TRANSPORT_SECURE_SOCKETS_H
+#define TRANSPORT_SECURE_SOCKETS_H
+
+/* standard includes. */
+#include <stdbool.h>
+/* Transport interface include. */
+#include "transport_interface.h"
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the transport interface implementation which uses
+ * Secure Sockets. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME     "Transport_Secure_Sockets"
+#endif
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_ERROR
+#endif
+
+#include "logging_stack.h"
+
+/**
+ * @brief TCP, TLS Connect / Disconnect return status.
+ */
+typedef enum TransportSocketStatus
+{
+    TRANSPORT_SOCKET_STATUS_SUCCESS = 0,         /**< Function successfully completed. */
+    TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER,   /**< At least one parameter was invalid. */
+    TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY, /**< Insufficient memory required to establish connection. */
+    TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, /**< Provided credentials were invalid. */
+    TRANSPORT_SOCKET_STATUS_API_ERROR,           /**< A call to a system API resulted in an internal error. */
```

Comment:
I think that use of `API_ERROR` could be confusing. We use `INTERNAL_ERROR` as the return code in the mbedtls based transport implementation file here: https://github.com/FreeRTOS/FreeRTOS/blob/lts-development/FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/tls_freertos.h#L134
```suggestion
    TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR,           /**< A call to a system API resulted in an internal error. */
```